txpool/txpool.go

```diff
@@ -789,3 +805,10 @@ func toHash(txs ...*types.Transaction) (hashes []types.Hash) {
 
 	return
 }
+
+// CanDeployContract checks if address can deploy smart contract
+func canDeployContract(from types.Address, whitelist []types.Address) bool {
```

Comment:
Since you defined the type `whiteList`, feel free to make this its method. Like:

```
func (w *whiteList) allowed(types.Address) bool
```

Reply:
Agreed with @zivkovicmilos that this should be a helper method https://github.com/0xPolygon/polygon-edge/pull/692#discussion_r951871861. Doesn't feel good to be whitelist method

Reply:
Fixed 6974e01cbb5bd4d4c82039d9a5decc59aba85c87

Reply:
@dbrajovic ping

Reply:
```
	if tx.IsContractCreation() && !p.deploymentWhitelist.allowed(tx.From) {
		return ErrSmartContractRestricted
	}
```

IMO, this read better and `canDeployContract` is not a free func, but rather a method.

Reply:
```
func (w *deploymentWhitelist) allowed(addr types.Address) bool {
	if len(w.addresses) == 0 {
		return true
	}
	
	_, ok := w.addresses[addr.String()]

	return ok
}
```

Reply:
Fixed e3da8b56708b58d84e64cd5dd87815f71c4bd019